### ISSUE TITLE
VDSO has missing dependencies

### ISSUE THREAD

**Issue by antonblanchard** [linuxppc/linux#30]
The VDSO is not getting rebuilt when swapping from an LE to a BE build:

```
make pseries_le_defconfig
make -j16 vmlinux
make pseries_defconfig
make -j16 vmlinux
...
/usr/bin/ld: arch/powerpc/kernel/vdso32/sigtramp.o: compiled for a little endian system and target is big endian
```

**Comment by mpe** [comment]
I can reproduce this. But it's not deterministic, sometimes it's `sigtramp.o`, sometimes `note.o` etc. If you run `file` on `arch/powerpc/kernel/vdso32/*.o` you see that some of the `.o`'s have been recompiled and some haven't.

**Comment by npiggin** [comment]
Should be fixed in mpe's tree

https://git.kernel.org/powerpc/c/b9a4a0d02c5b8d9a1397c11d74

**Comment by mpe** [comment]
Fixed by Nick in b9a4a0d02c5b8d9a1397c11d741d2a1a56381178.
